lib/std/json.zig

```diff
@@ -1379,3 +1420,34 @@ test "parsing empty string gives appropriate error" {
     defer p.deinit();
     testing.expectError(error.UnexpectedEndOfJson, p.parse(""));
 }
+
+test "escaped characters" {
+    const input =
+        \\{
+        \\  "backslash": "\\",
+        \\  "forwardslash": "\/",
+        \\  "newline": "\n",
+        \\  "carriagereturn": "\r",
+        \\  "tab": "\t",
+        \\  "formfeed": "\f",
+        \\  "backspace": "\b",
+        \\  "doublequote": "\"",
+        \\  "unicode": "\u0105"
```

Comment:
Could you add some tests with invalid escapes? e.g. `\x`,  an underlong unicode sequence `\u2abc`

Reply:
Invalid escapes are caught earlier in the parser and already tested in json/test.zig

Reply:
After checking, there are indeed some cases that are not caught. The tests only ensure that they don't crash the parser.

Reply:
Why is `\u2abc` problematic?

Reply:
I made the tests for json.StreamingParser run also for json.Parser.
That tests a lot of invalid stuff.

Reply:
> Why is `\u2abc` problematic?

Doh, I meant to make something up like `\u2xyz`.